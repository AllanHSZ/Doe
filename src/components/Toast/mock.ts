import { ToastProps } from '.'

export const ToastSuccess: ToastProps = {
  type: 'success',
  message: 'A simple success message'
}

export const ToastError: ToastProps = {
  type: 'error',
  message: 'A simple error message'
}
