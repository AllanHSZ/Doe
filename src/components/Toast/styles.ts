import styled, { css, DefaultTheme } from 'styled-components'
import { ToastTypes } from '.'

export const ToastWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    pointer-events: none;
    z-index: ${theme.layers.alwaysOnTop};
    padding-top: ${theme.spacings.medium};
    gap: ${theme.spacings.xsmall};
    position: absolute;
    overflow: hidden;
    max-width: min(36rem, 90vw);
    height: 100vh;
    width: 100vw;
    bottom: 0;
    right: 0;
    top: 0;

    @keyframes leftToRight {
      from {
        margin-left: 100%;
      }
      to {
        margin-left: 0;
      }
    }

    @keyframes rightToLeft {
      from {
        margin-left: 0;
      }
      to {
        margin-left: 100%;
      }
    }
  `}
`

export type WrapperProps = {
  hiding?: boolean
}

export const Wrapper = styled.div<WrapperProps>`
  ${({ hiding }) => css`
    display: flex;
    min-height: 8.6rem;
    width: 100%;
    min-width: min(36rem, 100%);
    animation: leftToRight 0.25s linear forwards;
    box-shadow: 0px 0px 8px 2px rgba(0, 0, 0, 0.15);

    ${!!hiding &&
    css`
      animation: leftToRight 0.25s linear forwards !important;
    `}
  `}
`

export type IconWrapperProps = {
  type: ToastTypes
  size: 'normal' | 'large'
}

const iconWrapperModifiers = {
  success: (theme: DefaultTheme) => css`
    svg {
      color: ${theme.colors.primary};
    }
  `,
  error: (theme: DefaultTheme) => css`
    svg {
      color: ${theme.colors.red};
    }
  `,
  informative: (theme: DefaultTheme) => css`
    svg {
      color: ${theme.colors.lightGray};
    }
  `,
  normal: (theme: DefaultTheme) => css`
    width: calc(${theme.spacings.large} * 2);

    svg {
      width: ${theme.spacings.large};
      height: ${theme.spacings.large};
    }
  `,
  large: (theme: DefaultTheme) => css`
    width: calc(${theme.spacings.xxlarge} * 2);

    svg {
      width: ${theme.spacings.xxlarge};
      height: ${theme.spacings.xxlarge};
    }
  `
}

export const IconWrapper = styled.div<IconWrapperProps>`
  ${({ theme, type, size }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    background: ${theme.colors.darkGray};
    border-top-left-radius: ${theme.border.radius.medium};
    border-bottom-left-radius: ${theme.border.radius.medium};
    width: 7rem;

    svg {
      width: ${theme.spacings.medium};
      height: ${theme.spacings.medium};
    }

    ${iconWrapperModifiers[type](theme)}
    ${iconWrapperModifiers[size](theme)}
  `}
`

export const Content = styled.div`
  ${({ theme }) => css`
    flex: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
    padding: ${theme.spacings.small};
    background: ${theme.colors.lightGray};
    gap: ${theme.spacings.small};
  `}
`

export const Title = styled.h4`
  ${({ theme }) => css`
    color: ${theme.colors.textEmphasis};
  `}
`

export const Message = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.text};
    font-size: ${theme.font.sizes.medium};
  `}
`
export const Action = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: flex-end;
    gap: ${theme.spacings.xxsmall};
  `}
`
