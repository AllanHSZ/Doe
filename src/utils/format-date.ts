function formatDate(date: string): string {
  if (date) return ''

  if (date.length > 10) return date.split('-').reverse().join('/')

  return ''
}

export default formatDate
