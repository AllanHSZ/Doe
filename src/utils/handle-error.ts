import { AxiosError } from 'axios'

const handlerError = (error: string | AxiosError): string => {
  console.log('handlerError:', error)

  if (typeof error === 'string') return error

  if (error?.response?.data?.error) return error?.response?.data?.error

  if (typeof error?.response?.data === 'string') return error?.response?.data

  return 'Erro desconhecido'
}

export default handlerError
