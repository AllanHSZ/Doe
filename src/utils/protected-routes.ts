import { GetServerSidePropsContext } from 'next'
import { DefaultSession, Session } from 'next-auth'
import { getSession } from 'next-auth/client'

export type MySession = {
  id: number
} & Pick<DefaultSession, 'expires' | 'user'>

async function protectedRoutes(
  context: GetServerSidePropsContext,
  session?: Session | null
) {
  if (!session) session = await getSession(context)

  if (!session?.user?.id) {
    context.res.writeHead(302, {
      Location: `/signin?callbackUrl=${context.resolvedUrl}`
    })
    context.res.end()
  }

  return session
}

export default protectedRoutes
