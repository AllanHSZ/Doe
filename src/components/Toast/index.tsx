import { Check, X } from 'react-feather'
import * as S from './styles'

export type ToastTypes = 'success' | 'error'

export type ToastProps = {
  type?: ToastTypes
  message: string
  hiding?: boolean
}

const Toast = ({ type = 'success', message }: ToastProps) => {
  const getIcon = (type: ToastTypes): React.ReactNode => {
    switch (type) {
      case 'success':
        return <Check />
      case 'error':
        return <X />
    }
  }

  return (
    <S.Wrapper>
      <S.IconWrapper type={type}>{getIcon(type)}</S.IconWrapper>
      <S.Content>
        <S.Message>{message}</S.Message>
      </S.Content>
    </S.Wrapper>
  )
}

export default Toast
