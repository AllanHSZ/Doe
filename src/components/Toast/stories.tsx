import { Story, Meta } from '@storybook/react/types-6-0'
import Toast from '.'

export default {
  title: 'Toast',
  component: Toast
} as Meta

export const Default: Story = () => <Toast />
