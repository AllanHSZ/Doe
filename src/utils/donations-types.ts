import { TextFieldOptions } from 'components/TextField'
// , ,
export default [
  {
    label: 'Acess√≥rio',
    value: 'ACESSORIO'
  },
  {
    label: 'Alimento',
    value: 'ALIMENTO'
  },
  {
    label: 'Roupa',
    value: 'ROUPA'
  }
] as TextFieldOptions[]
