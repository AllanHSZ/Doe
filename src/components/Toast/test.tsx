import { render, screen } from 'utils/test-utils'
import Toast from '.'

describe('<Toast />', () => {
  it('should render the heading', () => {
    const { container } = render(<Toast message="Teste message" />)

    expect(screen.getByText(/Teste message/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
