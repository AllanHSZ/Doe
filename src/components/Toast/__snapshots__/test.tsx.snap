// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toast /> should render the heading 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 6.4rem;
  width: 100%;
  min-width: min(30rem,100vw);
  -webkit-animation: leftToRight 0.25s linear forwards;
  animation: leftToRight 0.25s linear forwards;
  box-shadow: 0px 0px 8px 2px rgba(0,0,0,0.15);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #2E2F42;
  border-top-left-radius: 0.6rem;
  border-bottom-left-radius: 0.6rem;
  width: 7rem;
}

.c1 svg {
  width: 3.2rem;
  height: 3.2rem;
}

.c1 svg {
  color: #04D361;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 2.4rem 1.6rem;
  background: #F2F3F5;
}

.c3 {
  color: #394338;
  font-size: 1.4rem;
}

<div
  class="c0"
>
  <div
    class="c1"
    type="success"
  >
    <svg
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <polyline
        points="20 6 9 17 4 12"
      />
    </svg>
  </div>
  <div
    class="c2"
  >
    <p
      class="c3"
    >
      Teste message
    </p>
  </div>
</div>
`
